v19.2/delete.md

```diff
@@ -218,6 +244,72 @@ To sort and return deleted rows, use a statement like the following:
 (5 rows)
 ~~~
 
+### Delete on a specific index
+
+Suppose you create a multi-column index on the `users` table with the `name` and `city` columns.
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE INDEX ON users (name, city);
+~~~
+
+Now suppose you want to delete the two users named "Michael Brown". You can use the [`EXPLAIN (OPT)`](explain.html#opt-option) command to see how the [cost-based optimizer](cost-based-optimizer.html) decides to perform the delete:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> EXPLAIN (OPT) DELETE FROM users WHERE name='Michael Brown';
+~~~
+
+~~~
+                                 text
++---------------------------------------------------------------------+
+  delete users
+   └── scan users@users_name_city_idx
+        └── constraint: /8/7/6: [/'Michael Brown' - /'Michael Brown']
+(3 rows)
+~~~
+
+The output of the `EXPLAIN` statement shows that the optimizer scans the newly-created `users_name_city_idx` index when performing the delete. This makes sense, as you are performing a delete based on the `name` column.
+
+Now suppose that instead you want to perform a delete, but using the `id` column instead.
+
+{% include copy-clipboard.html %}
+~~~ sql
+> EXPLAIN (OPT) DELETE FROM users WHERE id IN ('70a3d70a-3d70-4400-8000-000000000016', '3d70a3d7-0a3d-4000-8000-00000000000c');
+~~~
+
+~~~
+                                                  text
++-------------------------------------------------------------------------------------------------------+
+  delete users
+   └── select
+        ├── scan users@users_name_city_idx
+        └── filters
+             └── id IN ('3d70a3d7-0a3d-4000-8000-00000000000c', '70a3d70a-3d70-4400-8000-000000000016')
+(5 rows)
+~~~
+
+The optimizer still scans the newly-created `users_name_city_idx` index when performing the delete. Although scanning the table on this index could still be the most efficient, you could want to assess the performance difference between using `users_name_city_idx` and an index on the `id` column, as you are performing a delete with a filter on the `id` column.
```

Comment:
I think this is easier to understand as 'may' rather than 'could', as in "you may want to assess the performance difference".